common/tzif.ksy

```diff
@@ -0,0 +1,279 @@
+          either be 0 or equal to `num_local_time_types`.
+      - id: num_leap_second_records
+        -orig-id: leapcnt
+        type: u4
+        doc: Number of leap-second records contained in the data block.
+      - id: num_transition_times
+        -orig-id: timecnt
+        type: u4
+        doc: Number of transition times contained in the data block.
+      - id: num_local_time_types
+        -orig-id: typecnt
+        type: u4
+        valid:
+          expr: >
+            _ != 0 and (num_is_ut_flags == 0 or _ == num_is_ut_flags) and
+            (num_is_std_flags == 0 or _ == num_is_std_flags)
+        doc: |
+          Number of local time type records contained in the data block. MUST
+          NOT be 0.
+      - id: len_tz_designations
+        -orig-id: charcnt
+        type: u4
+        valid:
+          expr: '_ != 0'
+        doc: |
+          Total number of octets used by the set of time zone designations
+          contained in the data block. MUST NOT be 0.
+    instances:
+      version:
+        value: 'ver_byte >= 0x32 ? ver_byte - 0x30 : 1'
+        doc: |
+          File format version number derived from `ver_byte`. The most recent
+          version as of March 2021 is 3.
+  tz_data:
+    doc-ref: https://tools.ietf.org/html/rfc8536#section-3.2
+    params:
+      - id: is_ts64
+        type: bool
+        doc: |
+          Chooses whether to use 64-bit or 32-bit time values. Set to `false`
+          for 32-bit time values (used in the version 1 data block). Set to
+          `true` for 64-bit time values (used in the version 2+ data block).
+    seq:
+      - id: transition_times
+        type:
+          switch-on: is_ts64
+          cases:
+            false: s4
+            true: s8
+        repeat: expr
+        repeat-expr: _parent.header.num_transition_times
+        doc: |
+          Array of POSIX time values sorted in ascending order. Each is used as
+          a transition time at which the rules for computing local time change.
+      - id: transition_types
+        type: u1
+        repeat: expr
+        repeat-expr: _parent.header.num_transition_times
+        doc: |
+          Array of transition type numbers. All but the last one map transition
+          times in the `transition_times` array to local time type records in
+          the `local_time_types` array. (The last one is present only for
+          consistency checking with the POSIX-style TZ string at the end of the
+          file.) The transition type numbers MUST be less than
+          `_parent.header.num_local_time_types` (the length of the
+          `local_time_types` array).
+      - id: local_time_types
+        type: tz_local_time_type
+        repeat: expr
+        repeat-expr: _parent.header.num_local_time_types
+        doc: Local time type records.
+      - id: tz_designations
+        size: _parent.header.len_tz_designations
+        valid:
+          expr: '_[_parent.header.len_tz_designations - 1] == 0'
```

Comment:
You can simplify this to

```suggestion
          expr: '_.last == 0'
```

..., see https://github.com/kaitai-io/kaitai_struct_tests/blob/97d68e9847aea0783eee21c27933ebf01fd4e441/formats/expr_bytes_ops.ksy#L16-L17.

One of the points in the 0.9 version was that we achieved a feature parity (in all languages except Perl) between "normal" arrays and byte arrays (i.e. implemented the [array methods](https://doc.kaitai.io/user_guide.html#_array_types) also for byte arrays - https://github.com/kaitai-io/kaitai_struct/issues/633), so you can take advantage of that.

Reply:
Thanks, I've changed it.